rankers/AggregateRanker/__init__.py

```diff
@@ -0,0 +1,32 @@
+from jina.executors.rankers import Chunk2DocRanker
+import numpy as np
+
+class AggregateRanker(Chunk2DocRanker):
+    """
+    :class:`AggregateRanker` calculates the score of the matched doc form the matched chunks. For each matched doc, the
+    score is aggregated from all the matched chunks belonging to that doc.
+    """
+
+    AGGREGATE_FUNCTIONS = ['min', 'max', 'mean', 'median', 'sum', 'prod']
+
+    def __init__(self, aggregate_function: str, is_reversed_score, *args, **kwargs):
+        """
+        :param: aggregate_function: defines the used aggregate function and can be one of:
+        [min, max, mean, median, sum, prod]
+        :param: is_reverse_score: True if a large score of the matched chunks mean a lower score of the matched doc,
+        othewise: False
+
+        """
+        super().__init__(*args, **kwargs)
+        self.is_reversed_score = is_reversed_score
+        self.aggregate_function = aggregate_function
+
+    def _get_score(self, match_idx, query_chunk_meta, match_chunk_meta, *args, **kwargs):
+        scores = match_idx[self.COL_SCORE]
+        if self.aggregate_function in self.AGGREGATE_FUNCTIONS:
+            aggregated_score = getattr(np, self.aggregate_function)(scores)
+        else:
+            raise ValueError(f'The aggregate function "{self.aggregate_function}" is not defined.')
+        if self.is_reversed_score:
+            aggregated_score = 1. / (1. + aggregated_score)
```

Comment:
If I am not mistaken, that does not work if we have positive and negative scores mixed. Having a test for this would be great.

While usually there are no negative scores, it can still happen.

Reply:
This function is for converting distance measures into similarity measures. Since distances can not be negative, we are fine - assuming the user is not using it for a similarity measure. To prevent misconfigurations, I think in the long term, we should put this conversion some levels up the call stack where we know what kind of metric was used.

Reply:
I think making it robust to have no `ZeroDivisionError` (and just have a warning could be good). In jina is desired that `drivers` tend to the best of the efforts not to know what the underlying executors are, this way we can split as much as possible the responsibilities and have different developer experiences.

Reply:
Okay, in that case, I would like to raise an exception even when the score is negative and `is_reversed_score` is used.
Otherwise around aggregated_score = -1, it will swap between low negative and high positive numbers. I think this will not be used on purpose.
![image](https://user-images.githubusercontent.com/11627845/98561606-625cd300-22a9-11eb-9603-6e0eece0363f.png)

Reply:
done